vlib/v/tests/cmp_between_alias_and_sum_type.vv

```diff
@@ -0,0 +1,10 @@
+struct Empty {}
+type IndexNumber = u32
+type NameIndexType = Empty | IndexNumber
+
+fn main() {
+	elem := NameIndexType(Empty{})
+	if elem is IndexNumber {
+
+	}
+}
```

Comment:
.vv files are not compiled by default, rename it to `_test.v` .

This is also not a good regression test, because it compiles fine on master too, i.e. it does not demonstrate what the PR fixes.